src/main/java/com/cburch/logisim/gui/prefs/WindowOptions.java

```diff
@@ -262,6 +269,10 @@ public void actionPerformed(ActionEvent e) {
         for (final var proj : nowOpen) {
           proj.getFrame().repaint();
         }
+      } else if (e.getActionCommand().equals(cmdSetAutoScaleFactor)) {
+        var tmp = AppPreferences.getAutoScaleFactor();
+        AppPreferences.SCALE_FACTOR.set(tmp);
+        zoomValue.setValue((int) (tmp * 100));
```

Comment:
Although this "resets" the scale factor once, it actually does not reset the value, as the new value is still written in the users preferences file. Hence the next time logisim is started it will take this "value" as scale factor instead of detecting the value as a fresh logisim install would do. The problem is that once the zoom-bar is used, a key is saved in the logisim java preferences file. The best solution would be to delete this key, but I do not know how to do this, and if it is even possible.

Reply:
I'm not sure what you mean. I moved the automatic calculation for SCALE_FACTOR to `getAutoScaleFactor()` (see the commit with AppPreferences.java). When you click the button, it sets SCALE_FACTOR to whatever that calculation would be.

Reply:
Exactly, and therefore a key in the java preferences file. Meaning that when you change to, for example, an external screen and start logisim you would need again need to press the button to "reset" to this screen. In case the button removes the key in the java preferences file, logisim would automatically "rescale" to the new screen, So the problem is that when "SCALE_FACTOR" is set once, it is reloaded each time as there is an entry for it in the java preferences file.

Reply:
Ok, I understand now. I'll have the button remove the preference, but I think there might be a cleaner way to solve that, as SCALE_FACTOR will save automatically during any session where the button isn't pressed.